demos/common/cpp/models/src/detection_model_retinaface_pt.cpp

```diff
@@ -0,0 +1,250 @@
+    }
+    InferenceEngine::InputInfo::Ptr& input = inputInfo.begin()->second;
+    std::string imageInputName = inputInfo.begin()->first;
+    inputsNames.push_back(imageInputName);
+    input->setPrecision(InferenceEngine::Precision::U8);
+    if (useAutoResize) {
+        input->getPreProcess().setResizeAlgorithm(InferenceEngine::ResizeAlgorithm::RESIZE_BILINEAR);
+        input->getInputData()->setLayout(InferenceEngine::Layout::NHWC);
+    }
+    else {
+        input->getInputData()->setLayout(InferenceEngine::Layout::NCHW);
+    }
+
+    //--- Reading image input parameters
+    imageInputName = inputInfo.begin()->first;
+    const InferenceEngine::TensorDesc& inputDesc = inputInfo.begin()->second->getTensorDesc();
+    netInputHeight = getTensorHeight(inputDesc);
+    netInputWidth = getTensorWidth(inputDesc);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
+
+    InferenceEngine::OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
+    landmarksNum = 0;
+
+    std::vector<uint32_t> outputsSizes[OT_MAX];
+    for (auto& output : outputInfo) {
+        output.second->setPrecision(InferenceEngine::Precision::FP32);
+        output.second->setLayout(output.second->getDims().size()==4 ? InferenceEngine::Layout::NCHW : InferenceEngine::Layout::CHW);
+        outputsNames.push_back(output.first);
+
+        outputsNames.resize(2);
+        if (output.first.find("bbox") != std::string::npos) {
+            outputsNames[OT_BBOX] = output.first;
+        }
+        else if (output.first.find("cls") != std::string::npos) {
+            outputsNames[OT_SCORES] = output.first;
+        }
+        else if (output.first.find("landmark") != std::string::npos) {
+            outputsNames.resize(std::max(outputsNames.size(), (size_t)OT_LANDMARK + 1));
+            outputsNames[OT_LANDMARK] = output.first;
+            landmarksNum = output.second->getDims()[2]/2; // Each landmark consist of 2 variables (x and y)
+        }
+        else {
+            continue;
+        }
+    }
+
+    if (outputsNames[OT_BBOX] == "" || outputsNames[OT_SCORES] == "") {
+        throw std::runtime_error("Bbox or cls layers are not found");
+    }
+
+    if (outputsNames.size() != 2 && outputsNames.size() != 3) {
+        throw std::logic_error("Expected 2 or 3 output blobs");
+    }
+
+    priors = generatePriorData();
+}
+
+
+std::vector<uint32_t> ModelRetinaFacePT::doThresholding(const InferenceEngine::MemoryBlob::Ptr& rawData, const float confidenceThreshold) {
+    std::vector<uint32_t> indicies;
+    auto desc = rawData->getTensorDesc();
+    auto sz = desc.getDims();
+    InferenceEngine::LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+
+    for (uint32_t x = 0; x < sz[1]; ++x) {
```

Comment:
`sz[1]` is `size_t`. Why is `indicies` a vector of `uint32_t`? To me it should be `size_t` as well.

Reply:
`indicies` is still a vector of `uint32_t`

Reply:
OK, looks I misunderstood the question. Max index value depends on priors size which is calculated by formula netWidth*netHeight*21/512. That gives us that to overflow uint_32 we need net size of 323598x323598 or greater - and that is definetely not our case. Same time, indicies array may have significant size, so changing uint32_t to size_t would double the RAM amount it consumes. sz[1] is size_t because it is returned by common function suitable for any network. So I suggest to keep indicies as vector of uint32_t to avoid useless overconsumption of RAM.

Reply:
> changing uint32_t to size_t would double the RAM amount it consumes

`size_t` is not grunted to be of size 64 bits.

If you cared that much about memory, you should have used `unit16_t`.

`indicies` is used for indexing and `size_t` is the type for indexing. I'm not sure that your `uint32_t` isn't casted to something similar to hardware equivalent of `size_t` when used, so what you may end up with is extra casts for `uint32_t` and even more, if it is performed by compiler than you will also have a casted copy of a value used: `uint32_t` and `size_t`. That will make things even worse for memory for the moment when the var is used (though I expect this not to happen in most cases).

Reply:
uint32 is CPU natively supported type unless you're going to run this on Z80. 
Just to avoid extra discussion: if you insist on this useless change, I'll do it, just to save time. Please answer here: do it or don't do it.

Reply:
Do it

Reply:
Done